src/cpu/reorder/simple_reorder.hpp

```diff
@@ -2294,6 +2294,119 @@ typename utils::enable_if<tag_i == format_tag::any &&
     }
 };
 
+template <SIMPLE_REORDER_TEMPL_DECL>
+struct simple_reorder_impl<SIMPLE_REORDER_TEMPL_CALL,
+typename utils::enable_if<tag_i == format_tag::any &&
+                          tag_traits_t<tag_o>::block_dims == bd::_AB &&
+                          type_i == data_type::u2 &&
+                          type_i == type_o>::type>
+{
+    static status_t is_applicable(const memory_desc_wrapper &input_d,
+                              const memory_desc_wrapper &output_d, const primitive_attr_t *attr) {
+        if (!(!input_d.has_runtime_dims_or_strides() &&
+             simple_attr_check(attr, false, true) &&
+             (order_keep ? output_d.matches_tag(tag_o) && input_d.is_plain()
+                         : input_d.matches_tag(tag_o) && output_d.is_plain())))
+            return status::invalid_arguments;
+
+        if (output_d.blocking_desc().inner_nblks != 3 ||
+            !utils::one_of(output_d.blocking_desc().inner_blks[2], 2, 4) ||
+            output_d.blocking_desc().inner_idxs[2] != 1)
+            return status::invalid_arguments;
+
+        return status::success;
+    }
+
+    GET_SCRATCHPAD_SIZE_ZERO();
+
+    static status_t execute(const cpu_reorder_pd_t *pd, const exec_ctx_t &ctx) {
+        DECLARE_COMMON_PARAMS();
+
+        int blksize_o = 1;
+        int blksize_i = 1;
+
+        for (int i = 0; i < output_d.blocking_desc().inner_nblks; i++) {
+            if (output_d.blocking_desc().inner_idxs[i] == 0)
+                blksize_o *= output_d.blocking_desc().inner_blks[i];
+            else
+                blksize_i *= output_d.blocking_desc().inner_blks[i];
+        }
+
+        const auto &dims = input_d.dims();
+        const auto &pdims = order_keep
+            ? output_d.padded_dims()
+            : input_d.padded_dims();
+
+        const int OC = dims[0];
+        const int NB_OC = pdims[0] / blksize_o;
+        const int IC = dims[1];
+        const int NB_IC = pdims[1] / blksize_i;
+
+        int i_mult_o = blksize_o;
+        int i_mult_i = blksize_i;
+
+        auto extract_2_bits = [&](uint8_t val, uint8_t idx) -> uint8_t {
```

Comment:
```suggestion
        auto extract_2_bits = [](uint8_t val, uint8_t idx) -> uint8_t {
```
Looks like we don't need to capture anything.

Reply:
Applied. Thanks Maksim!